packages/rocketchat-api/server/api.js

```diff
@@ -159,8 +194,24 @@ class API extends Restivus {
 						});
 
 						logger.debug(`${ this.request.method.toUpperCase() }: ${ this.request.url }`);
+						const requestIp = this.request.headers['x-forwarded-for'];
+						const objectForRateLimitMatch = {
+							IPAddr: requestIp,
+							route: `${ this.request.route }${ this.request.method.toLowerCase() }`
+						};
 						let result;
 						try {
+							const shouldVerifyRateLimit = rateLimiterDictionary.hasOwnProperty(objectForRateLimitMatch.route);
+							if (shouldVerifyRateLimit) {
+								rateLimiterDictionary[objectForRateLimitMatch.route].increment(objectForRateLimitMatch);
+								const attemptResult = rateLimiterDictionary[objectForRateLimitMatch.route].check(objectForRateLimitMatch);
+								if (!attemptResult.allowed) {
+									throw new Meteor.Error('error-too-many-requests', `Error, too many requests. Please slow down. You must wait ${ Math.ceil(attemptResult.timeToReset / 1000) } seconds before trying this endpoint again.`, {
```

Comment:
Can you set this value to the headers? Even when the request was a success? Take a look on the GitHub's api